### ISSUE TITLE
Expose Loss in Config

### ISSUE THREAD

**Issue by ebgoldstein** [dbuscombe-usgs/segmentation_zoo#22]
Options:
- [x] Dice Loss
- [x] Categorical cross entropy
- [x] hinge loss

**Comment by dbuscombe-usgs** [comment]
model part done:

```
if LOSS=='hinge':
    model.compile(optimizer = 'adam', loss =tf.keras.losses.Hinge(), metrics = [mean_iou, dice_coef])
elif LOSS=='dice'
    model.compile(optimizer = 'adam', loss =dice_coef_loss, metrics = [mean_iou, dice_coef])
elif LOSS.startswith('cat'):
    model.compile(optimizer = 'adam', loss =tf.keras.losses.CategoricalCrossentropy(), metrics = [mean_iou, dice_coef])
```

add `LOSS` to config. Options are `dice`, `hinge`, and `cat` (or any word starting with `cat`)

will now test all 3

**Comment by dbuscombe-usgs** [comment]
I believe Hinge(), not CategorcalHinge(), is appropriate here because labels are one-hot encoded

by the same logic, perhaps I should use BinaryCrossentropy() instead of CategoircalCrossentropy(

**Comment by dbuscombe-usgs** [comment]
nope ... "Use this crossentropy loss function when there are two or more label classes. We expect labels to be provided in a one_hot representation. If you want to provide labels as integers, please use SparseCategoricalCrossentropy loss."

**Comment by dbuscombe-usgs** [comment]
CategoricalHinge() is appropriate for one-hot encoded tensors https://keras.io/api/losses/hinge_losses/#categoricalhinge-function

**Comment by dbuscombe-usgs** [comment]
all 3 loss functions tested
